### PR TITLE
fix: [cds-43667]: fix notification configured for a user group

### PR DESCRIPTION
**Pull request by boopesh007**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38201)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/38201)

**Review by rijajoo — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201#-:-NDD7Jef89Wt9ckv9Zmi:bnfp4nl)* status: :shipit: complete! ✓

File: pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java [hunk 1]
```diff
@@ -184,13 +185,21 @@ private NotificationChannel getNotificationChannel(HarnessApprovalInstance insta
         String emailTemplateId = instance.isIncludePipelineExecutionHistory()
             ? PredefinedTemplate.HARNESS_APPROVAL_EXECUTION_NOTIFICATION_EMAIL.getIdentifier()
             : PredefinedTemplate.HARNESS_APPROVAL_NOTIFICATION_EMAIL.getIdentifier();
-        EmailConfigDTO emailConfigDTO = (EmailConfigDTO) notificationSettingConfig;
+        NotificationRequest.UserGroup.Builder notifyUserGroupBuilder =
+            NotificationRequest.UserGroup.newBuilder().setIdentifier(userGroup.getIdentifier());
```
Comment (rijajoo) [inline pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java:189]: if userGroup is not null, then ..

**Review by ershadalim — COMMENTED** [review]
Where are unit tests?

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201)* status: :shipit: complete! ✓

**Review by rijajoo — CHANGES_REQUESTED** [review]
Unit tests are failing.

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201#-:-NDSZJ9PCSrWmMw2VWXp:b-c6qg44)* status: :shipit: complete! ✓

___
*[`pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java` line 188 at r2](https://reviewable.io/reviews/harness/harness-core/38201#-NDSBWM_B3iDA8qFvMpU:-NDSBWM_B3iDA8qFvMpV:bzifj2g) ([raw file](https://github.com/harness/harness-core/blob/89b039251e9ef79b73770769fd19dea0399cadd2/pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java#L188)):*
> ```Java
>             ? PredefinedTemplate.HARNESS_APPROVAL_EXECUTION_NOTIFICATION_EMAIL.getIdentifier()
>             : PredefinedTemplate.HARNESS_APPROVAL_NOTIFICATION_EMAIL.getIdentifier();
>         NotificationRequest.UserGroup.Builder notifyUserGroupBuilder =
> ```

Please add userGroup != null check before this.

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201)* status: :shipit: complete! ✓

**Review by rijajoo — COMMENTED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201)* status: :shipit: complete! ✓

**Review by rijajoo — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201#-:-NDTKK5A-q0k4mDkeMKw:bnfp4nl)* status: :shipit: complete! ✓

**Review by rijajoo — APPROVED** [review]

**Comment by boopesh007** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java` line 188 at r2](https://reviewable.io/reviews/harness/harness-core/38201#-NDSBWM_B3iDA8qFvMpU:-NDaqjlRCBkTSG-WAhsm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/89b039251e9ef79b73770769fd19dea0399cadd2/pipeline-service/service/src/main/java/io/harness/pms/approval/notification/ApprovalNotificationHandlerImpl.java#L188)):*
<details><summary><i>Previously, rijajoo (Richa Jajoo) wrote…</i></summary><blockquote>

Please add userGroup != null check before this. 
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
trigger authorcheck

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38201)* status: 1. unresolved discussions.

**Pull request merged**